docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -481,19 +478,42 @@ particular, `TxsFront` is leaking implementation details outside the `Mempool` i
-parameters).
+In the current single-lane mempool, the function `ReapMaxBytesMaxGas(maxBytes, maxGas)` collects
+transactions in FIFO order from the CList until either reaching `maxBytes` or `maxGas` (both of
+these values are consensus parameters).
 
 With multiple CLists, we need to collect transactions from higher-priority lanes first, also in FIFO
 order, continuing with successive lanes in the `sortedLanes` array, that is, in decreasing priority
-order, and breaking the iteration when reaching `maxBytes` or `maxGas`. Similar to transaction
-dissemination, we want to avoid starvation of low-priority lanes by reaping transactions by their
-weight and fairly. Consequently, we should implement a scheduling algorithm that meets these
-properties, which could be the same or similar to the one used for transaction dissemination.
-
-The mempool is locked during `ReapMaxBytesMaxGas`, so no transaction will be added or removed from
-the mempool during reaping.
+order, and breaking the iteration when reaching `maxBytes` or `maxGas`. Note that the mempool is
+locked during `ReapMaxBytesMaxGas`, so no transaction will be added or removed from the mempool
+during reaping.
+
+This simple algorithm, though good enough for an MVP, does not guarantee that low-priority lanes
+will not starve. That is why we prefer to implement one that is starvation-free, as explained in the
+next section. It could be the same algorithm or similar to the one used for transaction
+dissemination.
+
+### Prioritization logic
+
+For transaction dissemination and for reaping transactions for creating blocks we want a scheduling
+algorithm that satisfies the properties "Priorities between classes" and "FIFO ordering per class".
+This means that it must support selection by _weight_, ensuring each lane gets a fraction of the P2P
+channel capacity proportional to its priority. Additionally, we want the algorithm to be _fair_ to
+prevent starvation of low-priority lanes. 
+
+A first option that meets these criteria is the current prioritization algorithm on the P2P reactor,
+which we could easily reimplement in the mempool. It works as follows:
+- On each P2P channel, the variable `recentlySent` keeps track of how many bytes were recently sent
+  over the channel. Every time data is sent, increase `recentlySent` with the number of bytes
+  written to the channel. Every 2 seconds, decrease `recentlySent` by 20% on all channels (these
+  values are fixed).
+- When sending the next message, [pick the channel][selectChannelToGossipOn] whose ratio
+  `recentlySent/Priority` is the least.
+
+From the extensive research in operating systems and networking, we can choose for the MVP an
+existing scheduling algorithm that meet these requirements and are straightforward to implement,
+such as a variant of the [Weighted Round Robin][wrr] (WRR) algorithm. Another one to consider is
+[Weighted Random Early Detection][wred].
```

Comment:
```suggestion
such as a variant of the [Weighted Round Robin][wrr] (WRR) algorithm.
```

I was wrong w.r.t. WRED, it's rather used to help congestion control algorithms (in the presence of QoS), so not really related here.